packages/grid/_modules_/grid/components/columnHeaders/ColumnHeaderFilterIcon.tsx

```diff
@@ -25,17 +21,20 @@ export function ColumnHeaderFilterIcon(props: ColumnHeaderFilterIconProps) {
       event.preventDefault();
       event.stopPropagation();
 
-      const { open, openedPanelValue } = preferencePanel;
+      const { open, openedPanelValue } = gridPreferencePanelStateSelector(
+        apiRef!.current.getState(),
+      );
+
       if (open && openedPanelValue === GridPreferencePanelsValue.filters) {
         apiRef!.current.hideFilterPanel();
       } else {
         apiRef!.current.showFilterPanel();
       }
     },
-    [apiRef, preferencePanel],
+    [apiRef],
   );
 
-  if (!counter || options.disableColumnFilter) {
```

Comment:
Why do we need to remove it from here? I guess it is because the parent component is memoized.

Reply:
It was suggested in https://github.com/mui-org/material-ui-x/pull/1628#discussion_r630284135